packages/pluggableWidgets/combobox-web/src/helpers/BaseOptionsProvider.ts

```
@@ -0,0 +1,59 @@
+import { FilterTypeEnum } from "typings/ComboboxProps";
+import { CaptionsProvider, OptionsProvider, Status } from "./types";
+import { MatchSorterOptions, matchSorter } from "match-sorter";
+import { getFilterTypeOptions } from "./utils";
+
+export class BaseOptionsProvider<T = unknown, P = object> implements OptionsProvider<T, P> {
+    protected options: string[] = [];
+    private caption: CaptionsProvider;
+    private trigger?: () => void;
+
+    filterType: FilterTypeEnum = "contains";
+    searchTerm = "";
+
+    constructor(caption: CaptionsProvider) {
+        this.caption = caption;
+    }
+
+    get hasMore(): boolean {
+        return false;
+    }
+
+    get status(): Status {
+        return "available";
+    }
+
+    getAll(): string[] {
+        const matchSorterOptions: MatchSorterOptions<string> = {
+            keys: [v => this.caption.get(v)],
+            ...getFilterTypeOptions(this.filterType)
+        };
+
+        if (this.searchTerm === "") {
+            matchSorterOptions.sorter = option => option;
+        }
+
+        return matchSorter(this.options, this.searchTerm || "", matchSorterOptions);
+    }
+
+    setSearchTerm(term: string): void {
+        this.searchTerm = term;
+        this.trigger?.();
+    }
+
+    onAfterSearchTermChange(callback: () => void): void {
+        this.trigger = callback;
+    }
+
+    loadMore(): void {
+        return undefined;
+    }
+
+    _updateProps(_props: P): void {}
+    _optionToValue(_option: string | null): T | undefined {
+        return undefined;
```

this should throw an exception, not implemented